packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -345,17 +357,22 @@ abstract contract CrossDomainMessenger is
             "CrossDomainMessenger: message has already been relayed"
         );
 
+        // If there is not enough gas left to perform the external call and finish the execution,
+        // return early and assign the message to the failedMessages mapping.
         // If `xDomainMsgSender` is not the default L2 sender, this function
         // is being re-entered. This marks the message as failed to allow it
         // to be replayed.
-        if (xDomainMsgSender != Constants.DEFAULT_L2_SENDER) {
+        if (
+            gasleft() < _minGasLimit + RELAY_RESERVED_GAS ||
```

Comment:
I'm slightly worried about this breaking gas estimation

Reply:
Yeah, this totally breaks gas estimation, good spot. Adding a revert if the `tx.origin` is the estimation address.